### PR TITLE
We have a muscle-controlled cyclist!

### PR DESCRIPTION
**Pull request by brocksam**
Likes to steer to the right a bit too much. Potentially a bug in the geometry of the musculotendons, or it could be that the initial positive roll rate interacts with the delay in the (very simple) controller in a way that means our cyclist can't damp out the initial perturbation. I think an improved controller will help here, but this is good enough for now, it at least shows that they rider can stay upright on the bike.

### CODE REVIEW

File: examples/bicycle_muscle_steer.py [hunk 1]
```diff
@@ -749,15 +749,37 @@ def __init__(self, *args, **kwargs):
...
+
+    We want the right bicep and left tricep to excite when the roll rate is
+    positive, and the left bicep and right tricep to excite when the roll
+    rate is negative.
+
+    """
+    max_roll_rate = 0.2
+    if roll_rate > 0.0:
+        normalized_roll_rate = roll_rate / max_roll_rate
+        e_bicep_r = normalized_roll_rate if normalized_roll_rate < 0.99 else 0.99
```
Comment (TJStienstra) [inline examples/bicycle_muscle_steer.py:764]: I think that it is more efficient to use `e_bicep_r = min((normalized_roll_rate, 0.99))`

Comment (brocksam) [reply examples/bicycle_muscle_steer.py:764]: I agree, that much cleaner. Jason and I were rushing to try and get this running before lunch so there are a few horrors like this in this script.

File: examples/bicycle_muscle_steer.py [hunk 2]
```diff
@@ -749,15 +749,37 @@ def __init__(self, *args, **kwargs):
...
+        e_bicep_l = 0.01
+        e_tricep_r = 0.01
+        e_tricep_l = normalized_roll_rate if normalized_roll_rate < 0.99 else 0.99
+    else:
+        normalized_roll_rate = -roll_rate / max_roll_rate
+        e_bicep_r = 0.01
+        e_bicep_l = normalized_roll_rate if normalized_roll_rate < 0.99 else 0.99
+        e_tricep_r = normalized_roll_rate if normalized_roll_rate < 0.99 else 0.99
+        e_tricep_l = 0.01
```
Comment (brocksam) [inline examples/bicycle_muscle_steer.py:773]:
```suggestion
        normalized_roll_rate = roll_rate / max_roll_rate
        e_bicep_r = min(normalized_roll_rate, 0.99)
        e_bicep_l = 0.01
        e_tricep_r = 0.01
        e_tricep_l = min(normalized_roll_rate, 0.99)
    else:
        normalized_roll_rate = -roll_rate / max_roll_rate
        e_bicep_r = 0.01
        e_bicep_l = min(normalized_roll_rate, 0.99)
        e_tricep_r = min(normalized_roll_rate, 0.99)
        e_tricep_l = 0.01
```

**Comment by moorepants** [comment]
I pulled it and it wasn't stable for me, but I did a little refactoring and just have it checking whether a simple pulse in muscle activation causes counter steering and it does. I've pushed some changes to this branch.

**Comment by moorepants** [comment]
Here it is:

stable-muscle-steering



https://github.com/brocksam/sympy-biomechanics/assets/276007/9a4c4900-0717-45a4-8267-5ffeee0cd3f7

**Comment by moorepants** [comment]
I had to swap the sign on the control (still confused about that). Need to investigate.

**Comment by moorepants** [comment]
Merging this, as it runs.

**Pull request merged**